### PR TITLE
cluster tests: Use REFRESH EVERY with blue/green deployment

### PR DESCRIPTION
**Pull request by def-**
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: test/cluster/blue-green-deployment/deploy.td [hunk 1]
```diff
@@ -9,8 +9,10 @@
 
 > CREATE SCHEMA prod_deploy
 > CREATE MATERIALIZED VIEW prod_deploy.counter_mv IN CLUSTER prod_deploy AS SELECT count(*), 'some new value' FROM counter
+> CREATE MATERIALIZED VIEW prod_deploy.counter_mv2 IN CLUSTER prod_deploy WITH (REFRESH EVERY '60 seconds') AS SELECT count(*), 'some new value' FROM counter
 > CREATE DEFAULT INDEX IN CLUSTER prod_deploy ON counter
 > CREATE DEFAULT INDEX IN CLUSTER prod_deploy ON prod_deploy.counter_mv
+> CREATE DEFAULT INDEX IN CLUSTER prod_deploy ON prod_deploy.counter_mv2
```
Comment (ggevay) [inline test/cluster/blue-green-deployment/deploy.td:15]: Does this test involve restarts? If yes, then please don't create indexes on REFRESH MVs for now, due to the problem explained in the "Temporary Limitations" section of the original PR https://github.com/MaterializeInc/materialize/pull/23819

Comment (def-) [reply test/cluster/blue-green-deployment/deploy.td]: It doesn't, but removed anyway.

File: test/cluster/blue-green-deployment/deploy.td [hunk 2]
```diff
@@ -9,8 +9,10 @@
 
 > CREATE SCHEMA prod_deploy
 > CREATE MATERIALIZED VIEW prod_deploy.counter_mv IN CLUSTER prod_deploy AS SELECT count(*), 'some new value' FROM counter
+> CREATE MATERIALIZED VIEW prod_deploy.counter_mv2 IN CLUSTER prod_deploy WITH (REFRESH EVERY '60 seconds') AS SELECT count(*), 'some new value' FROM counter
```
[the remark this answers is not in the document]
Comment (def-) [reply test/cluster/blue-green-deployment/deploy.td:12]: I mostly wanted to see if the readiness query would be ok with slowly refreshing materialized views, or if they would count as having a local_lag > 1s.

File: test/cluster/blue-green-deployment/deploy.td (hunk 1 above)
Comment (ggevay) [reply test/cluster/blue-green-deployment/deploy.td]: Thx!

**Review by ggevay — APPROVED** [review]

**Review by teskje — APPROVED** [review]
Thanks!

**Review by nrainer-materialize — APPROVED** [review]

**Pull request merged**